File: IT/2/Database/Esercizi/Java/Esercizio1.java
/*
file Esercizio1.java
eseguire con: java -cp .:sqlitejdbc-v<nnn>.jar Esercizio1
*/

import java.sql.DriverManager;
import java.sql.Connection;
import java.sql.Statement;
import java.sql.SQLException;

public class Esercizio1 {
	public static void main(String [] args) {
		Connection conn = null;
		Statement stmt = null;

		try {
			Class.forName("org.sqlite.JDBC");
			String dbName = "students.db";
			conn = DriverManager.getConnection("jdbc:sqlite:" + dbName);
			stmt = conn.createStatement();

			stmt.executeUpdate("CREATE TABLE students (id INTEGER PRIMARY KEY, number VARCHAR, name VARCHAR, surname VARCHAR, age INTEGER)");

			System.out.println("Creata tabella degli studenti.");
		}
		catch (ClassNotFoundException e) {
			System.err.println(e);
		}
		catch (SQLException e) {
			System.err.println(e);
		}
		finally {
			try {
				if (stmt != null) stmt.close();
				if (conn != null) conn.close();
			}
			catch (SQLException e) {
				System.err.println(e);
			}
		}
	}
}

File: IT/2/Database/Esercizi/Python/esercizio1.py
# file esercizio1.py

from __future__ import print_function
import sqlite3

conn = sqlite3.connect("students.db")
cursor = conn.cursor()

try:
	cursor.execute("CREATE TABLE students (id INTEGER PRIMARY KEY, number VARCHAR, name VARCHAR, surname VARCHAR, age INTEGER)")
	conn.commit()
	print("Creata tabella degli studenti.")

except sqlite3.OperationalError as e:
	print(e)

